### PR TITLE
fix: connectivity from metadata preserves excluded connections

### PR DESCRIPTION
**Pull request by fSosaRey**

### CODE REVIEW

File: src/core/connectivity.jl [hunk 1]
```diff
@@ -63,6 +63,18 @@ struct LineConnectivity <: AbstractConnectivity
...
+function Base.isequal(c0::LineConnectivity, c1::LineConnectivity)::Bool
+    if c0.dimension != c1.dimension
+        return false
+    end
+
+    if c0.excluded_connections != c1.excluded_connections
+        return false
+    end
+
+    return c0.excluded_positions == c1.excluded_positions
```
Comment (npoirie1) [inline src/core/connectivity.jl:75]: The `excluded_positions` are not sorted. They should be sorted in the `LineConnectivity` constructor.

File: src/core/connectivity.jl [hunk 2]
```diff
@@ -253,6 +265,18 @@ struct LatticeConnectivity <: AbstractConnectivity
...
+function Base.isequal(c0::LatticeConnectivity, c1::LatticeConnectivity)::Bool
+    if c0.dimensions != c1.dimensions
+        return false
+    end
+
+    if c0.excluded_connections != c1.excluded_connections
+        return false
+    end
+
+    return c0.excluded_positions == c1.excluded_positions
```
Comment (npoirie1) [inline src/core/connectivity.jl:277]: The `excluded_positions` aren't sorted for the `LatticeConnectivity`.

File: test/qpu_interface.jl [hunk 3]
```diff
@@ -511,6 +511,72 @@ end
...
+        Snowflurry.with_excluded_positions(LineConnectivity(4), [1, 2, 3]),
+    )
+    @test !isequal(
+        Snowflurry.with_excluded_positions(LineConnectivity(4), [1, 2, 3]),
+        Snowflurry.with_excluded_positions(LineConnectivity(4), [4]),
+    )
+    @test isequal(
+        Snowflurry.with_excluded_positions(LineConnectivity(4), [1, 2, 3]),
+        Snowflurry.with_excluded_positions(LineConnectivity(4), [1, 2, 3]),
+    )
```
Comment (npoirie1) [inline test/qpu_interface.jl:530]: Add a test with unsorted excluded positions.

File: test/qpu_interface.jl [hunk 4]
```diff
@@ -511,6 +511,72 @@ end
...
+        Snowflurry.with_excluded_connections(LineConnectivity(4), [(1, 2)]),
+    )
+    @test !isequal(
+        Snowflurry.with_excluded_connections(LineConnectivity(4), [(1, 2), (2, 3)]),
+        Snowflurry.with_excluded_connections(LineConnectivity(4), [(1, 2)]),
+    )
+    @test isequal(
+        Snowflurry.with_excluded_connections(LineConnectivity(4), [(1, 2), (2, 3)]),
+        Snowflurry.with_excluded_connections(LineConnectivity(4), [(1, 2), (2, 3)]),
+    )
```
Comment (npoirie1) [inline test/qpu_interface.jl:543]: Could add a test with unsorted excluded connections.

File: test/qpu_interface.jl [hunk 5]
```diff
@@ -511,6 +511,72 @@ end
...
+        Snowflurry.with_excluded_positions(LatticeConnectivity(4, 3), [1, 2, 3]),
+    )
+    @test !isequal(
+        Snowflurry.with_excluded_positions(LatticeConnectivity(4, 3), [1, 2, 3]),
+        Snowflurry.with_excluded_positions(LatticeConnectivity(4, 3), [4]),
+    )
+    @test isequal(
+        Snowflurry.with_excluded_positions(LatticeConnectivity(4, 3), [1, 2, 3]),
+        Snowflurry.with_excluded_positions(LatticeConnectivity(4, 3), [1, 2, 3]),
+    )
```
Comment (npoirie1) [inline test/qpu_interface.jl:562]: Add a test with unsorted excluded positions

File: test/qpu_interface.jl [hunk 6]
```diff
@@ -511,6 +511,72 @@ end
...
+        Snowflurry.with_excluded_connections(LatticeConnectivity(4, 3), [(1, 4)]),
+    )
+    @test !isequal(
+        Snowflurry.with_excluded_connections(LatticeConnectivity(4, 3), [(1, 4), (2, 4)]),
+        Snowflurry.with_excluded_connections(LatticeConnectivity(4, 3), [(1, 4)]),
+    )
+    @test isequal(
+        Snowflurry.with_excluded_connections(LatticeConnectivity(4, 3), [(1, 4), (2, 4)]),
+        Snowflurry.with_excluded_connections(LatticeConnectivity(4, 3), [(1, 4), (2, 4)]),
+    )
```
Comment (npoirie1) [inline test/qpu_interface.jl:575]: Could add a test with unsorted excluded connections.

File: test/qpu_interface.jl [hunk 7]
```diff
@@ -1510,6 +1576,70 @@ end
...
+        throw(ErrorException("request checker should not be called during test"))
+    end
+
+    for (qpu_name, QPU, metadata, expected_connectivity) in test_specs
+
+        test_client = Client(
+            host = expected_host,
+            user = expected_user,
+            access_token = expected_access_token,
+            requestor = requestor = MockRequestor(
```
Comment (npoirie1) [inline test/qpu_interface.jl:1627]: Typo (requestor = requestor = ...)

**Review by npoirie1 — CHANGES_REQUESTED** [review]

File: test/qpu_interface.jl (hunk 4 above)
Comment (fSosaRey) [reply test/qpu_interface.jl:547]: added

File: test/qpu_interface.jl (hunk 5 above)
Comment (fSosaRey) [reply test/qpu_interface.jl:574]: added

File: test/qpu_interface.jl (hunk 6 above)
Comment (fSosaRey) [reply test/qpu_interface.jl:591]: added

File: test/qpu_interface.jl (hunk 7 above)
Comment (fSosaRey) [reply test/qpu_interface.jl:1643]: goot catch! puzzled as to why it doesn't crash

File: test/qpu_interface.jl (hunk 3 above)
Comment (fSosaRey) [reply test/qpu_interface.jl:530]: added

File: src/core/connectivity.jl (hunk 2 above)
Comment (fSosaRey) [reply src/core/connectivity.jl]: refactored the isequal for a new type ExcludedPositions

File: src/core/connectivity.jl (hunk 1 above)
Comment (fSosaRey) [reply src/core/connectivity.jl]: added

File: test/qpu_interface.jl (hunk 6 above)
Comment (npoirie1) [reply test/qpu_interface.jl:591]: You actually forgot to add the test.

Comment (fSosaRey) [reply test/qpu_interface.jl:591]: haha yes, ok it's added now

**Review by npoirie1 — APPROVED** [review]

**Pull request merged**